docs/resources/server/enchantments/index.md

```diff
@@ -0,0 +1,307 @@
+            "type": "minecraft:linear",
+            "base": 1,
+            "per_level_above_first": 1
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+### Enchantment Costs
+
+The `max_cost` and `min_cost` fields specify boundaries for how much enchanting power is needed to create this enchantment. There is a somewhat convoluted procedure to actually make use of these values, however.
+
+First, the table takes into account the return value of `IBlockStateExtension#getEnchantPowerBonus()` for the surrounding blocks. From this, it derives a 'base cost' for each slot. This cost is shown in-game as the green numbers besides the enchantments in the menu. For each enchantment, the base cost is modified twice by a random value derived from the item's enchantability (its return value from `IItemStackExtension#getEnchantmentValue()`), like so:
+
+`(Modified Cost) = (Base Cost) + random.nextInt(e / 4 + 1) + random.nextInt(e / 4 + 1)`, where `e` is the enchantability score.
+
+This modified cost is adjusted up or down by a random 15%, and then is finally used to choose an enchantment. 
+
+In practical terms, this means that the cost values in your enchantment definition might be above 30, sometimes far above. For example, with an enchantability 10 item, the table could produce enchantments up to 1.15 * (30 + 2 * (10 / 4) + 1) = 40 cost. 
+
+## Enchantment Effect Components
+
+Enchantment effect components are specially-registered [Data Components] that determine how an enchantment functions. The type of the component defines its effect, while the data it contains is used to inform or modify that effect. For instance, the `minecraft:damage` component modifies the damage that a weapon deals by an amount determined by its data.
+
+Vanilla defines various [built-in enchantment effect components], which are used to implement all vanilla enchantments.
+
+### Custom Enchantment Effect Components
+
+Enchantment effect component types must be [registered] to `BuiltInRegistries.ENCHANTMENT_EFFECT_COMPONENT_TYPE`, which takes a `DataComponentType<?>`. For example, you could register an enchantment effect component that can store an `ExampleEffectData` object as follows:
+
+```java
+// In some registration class
+public static final DeferredRegister<DataComponentType<?>> ENCHANTMENT_COMPONENT_TYPES = DeferredRegister.create(BuiltInRegistries.ENCHANTMENT_EFFECT_COMPONENT_TYPE, "examplemod");
+
+public static final DeferredHolder<DataComponentType<?>, DataComponentType<ExampleEffectData>>> EXAMPLE =
+    ENCHANTMENT_COMPONENT_TYPES.register("example",
+        () -> DataComponentType.<ExampleEffectData>builder()
+            .persistent(ExampleEffectData.CODEC)
+            .build());
+```
+
+There are no inheritance requirements on the data held by an enchantment effect component.
+
+### `ConditionalEffect`
+Wrapping the type in `ConditionalEffect<?>` allows the enchantment effect component to take effect based on conditions informed by a [LootContext]. 
```

Comment:
I don't understand this sentence. I understand you're trying to say the `LootItemCondition` is tested with the provided `LootContext` the enchanted object is used in, but it is worded strangely as it's not 'informed' but rather 'given' the context to work within.